packages/itwin/tree-widget/src/tree-widget-react/components/trees/models-tree/internal/ModelsTreeIdsCache.ts

```diff
@@ -266,6 +267,41 @@ export class ModelsTreeIdsCache {
     }
   }
 
+  private async *queryModeledElements() {
+    const query = `
+      SELECT
+        pe.ECInstanceId modeledElementId,
+        pe.Category.Id categoryId,
+        pe.Model.Id modelId
+      FROM
+        BisCore.Model m JOIN ${this._hierarchyConfig.elementClassSpecification} pe ON pe.ECInstanceId = m.ModeledElement.Id
+    `;
+    for await (const row of this._queryExecutor.createQueryReader({ ecsql: query }, { rowFormat: "ECSqlPropertyNames", limit: "unbounded" })) {
+      yield { modelId: row.modelId, categoryId: row.categoryId, modeledElementId: row.modeledElementId };
+    }
+  }
+
+  private async getModelWithCategoryModeledElements() {
+    this._modelWithCategoryModeledElements ??= (async () => {
+      const modelWithCategoryModeledElements = new Map<Id64String, Id64Set>();
+      const getKey = (modelId: Id64String, categoryId: Id64String) => `${modelId}-${categoryId}`;
```

Comment:
I think it would be more efficient to just create the key once in `for await` loop